apps/pipelines/logging.py

```diff
@@ -15,42 +15,43 @@ class PipelineLoggingCallbackHandler(BaseCallbackHandler):
     def __init__(self, pipeline_run, verbose=False) -> None:
         self.pipeline_run = pipeline_run
         self.logger = get_logger(uuid.uuid4().hex, self.pipeline_run)
-        self.depth = 0
         self.verbose = verbose
         self._run_id_names = {}
 
     def _should_log(self, kwargs):
+        """We only log steps that are defined nodes inside of the pipeline.
+
+        The nodes themselves should handle logging further if more information is required.
+        """
+
         if "langsmith:hidden" in kwargs.get("tags", []):
             return False
+
         if self.verbose:
             return True
+
         if kwargs.get("run_id") in self._run_id_names:
             return True
-        from apps.pipelines.nodes import nodes
 
-        if hasattr(nodes, kwargs.get("name", "")):
+        if kwargs.get("name") in self.pipeline_run.pipeline.node_ids:
             self._run_id_names[kwargs.get("run_id")] = kwargs.get("name")
             return True
+
         return False
 
     def on_chain_start(self, serialized, inputs, *args, **kwargs):
-        self.depth += 1
-        from apps.pipelines.models import PipelineRunStatus
-
-        self.pipeline_run.status = PipelineRunStatus.RUNNING
         if self._should_log(kwargs):
-            self.logger.info(f"{kwargs.get('name', serialized.get('name'))} starting")
+            self.logger.info(
```

Comment:
I have been making some changes to this code in [this PR](https://github.com/dimagi/open-chat-studio/pull/474). Logging of the node start + end happens within the nodes itself now, since they are not langchain runnables anymore, but pure functions. See [this](https://github.com/dimagi/open-chat-studio/pull/474/commits/91ef3d203f5ec4255425524a4b0b9e17e595e515) and [this](https://github.com/dimagi/open-chat-studio/pull/474/commits/2de42df142973be461d11ce90b28830f2db8fc22) change (oh and this [one](https://github.com/dimagi/open-chat-studio/pull/474/commits/259d3ed3fad52521abd25a94f6d9aeaf47d842c0) as well). I found it easier to work with if the nodes are not runnables.

Reply:
Commented on that PR directly - it looks great!